SubZero/src/model/filter/Filter.h

```diff
@@ -7,11 +7,42 @@
 
 #ifndef FILTER_H_
 #define FILTER_H_
+#include "DataDefinition.h"
+#include "RGBFilter.h"
+//#include "HSVFilter.h"
+
 
 class Filter {
+	/*
+	 * Filter is a virtual class that is further specified into specific
+	 * filters accessible by FilterFactory. In general filters will have
+	 * a filter() function that will filter input according to the
+	 * algorithm it contains and the parameters it holds.
+	 */
+
 public:
 	Filter();
 	virtual ~Filter();
+
+	// The filter function takes a pointer and applies the algorithm of the
+	// specific filter to the content of the pointer. The content of the
+	// pointer is altered and 0 is returned for successful filter, 1 for no
+	// action performed due to incorrect type, 2 for other errors.
+	virtual int filter(CamData * inputImg) {
+		return 1;
+	}
+	// Overload for FPGA filtering
+	virtual int filter(FPGAData * inputAttitude) {
+		return 1;
+	}
+
+	// Get the ID of the specific filter instance
+	std::string getID() {
+		return ID;
+	}
+private:
+	std::string ID;
+
```

Comment:
You most likely explained the ID concept during meeting already. Sorry I missed out. Just from the naming and comments here I can't figure out how the ID works though. When do two filter instances have the same ID? Same type and same parameters? Or just same type? E.g. Will a gaussian blur filter with sigma s and dimension d have the same ID as one with s' and d' ?